core/pom.xml

```diff
@@ -39,19 +40,14 @@
             <version>3.3.0.Final</version>
         </dependency>
         <dependency>
-            <groupId>org.jboss.spec.javax.batch</groupId>
-            <artifactId>jboss-batch-api_1.0_spec</artifactId>
-            <scope>provided</scope>
-        </dependency>
-        <dependency>
             <groupId>javax.inject</groupId>
             <artifactId>javax.inject</artifactId>
```

Comment:
Ok, you're right, it seems there is no such thing as transitive provided dependencies, even just when compiling. Which makes sense, I guess. So yes, we need this explicit "provided" dependency to make the build pass.